### PR TITLE
fix some compiler warnings pt1

### PR DESCRIPTION
**Pull request by adizaimi**
fix some compiler warnings pt1

### CODE REVIEW

File: mem/mem.c [hunk 1]
```diff
@@ -384,6 +377,7 @@ int comdb2ma_init(size_t init_sz, size_t max_cap)
              * 3. If a thread is not allowed to create a mspace, it will
              *    be uniformly assigned one (to minimize contention).
              */
+            char *env_mspace_max;
```
Comment (riverszhang89) [inline mem/mem.c]: What compiler warning do you see here?

File: mem/mem.c [hunk 2]
```diff
@@ -2090,6 +2082,13 @@ static void pfx_ctrace(const char *format, ...)
 
 #if !defined(USE_SYS_ALLOC) && !defined(COMDB2MA_OMIT_BMEM)
 /* COMDB2 BLOCKING MEMORY ALLOCATOR { */
+
+/* the pthread key is shared amongst all blocking allocators */
```
Comment (riverszhang89) [inline mem/mem.c]: I specifically placed these lines in "static code" section. Is this causing a compiler warning?

File: mem/mem.c (hunk 1 above)
Comment (adizaimi) [reply mem/mem.c]: it's an unused variable warning -- only used if PER_THREAD_MALLOC is defined.

Comment (riverszhang89) [reply mem/mem.c]: Gotcha.

File: mem/mem.c (hunk 2 above)
Comment (adizaimi) [reply mem/mem.c]: Yes, i put them back where you had them, and surrounded them with appropriate ifdef.
Thanks

**Pull request merged**